### PR TITLE
Security Solution Upgrade testing Phase 1: Prepare the data that we are going to use for testing

### PR DESCRIPTION
**Pull request by MadameSheema**
Following the issue: https://github.com/elastic/elastic-stack-testing/issues/928 we are preparing the test data we are going to need to perform the testing.

The idea is first to create a custom query detection rule, and then, generate an alert for it.

In order to do that, we are creating a new `UploadSecuritySolutionData.java`. Here we are first creating a siem signals index for the default workspace. This is because, in order to be able to create a rule, we need that index to exist in the workspace where the rule is going to be created. Once the index is created, we create a custom query detection rule and then we create a document that matches with the rule index and query to generate the alert.

As part of this PR, we are also creating the `upload_security_solution_data` task that executes the above.

### CODE REVIEW

**Comment by MadameSheema** [comment]
@liza-mae cam you please take a look at this PR? Thanks :)

File: ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            }
+        }
+    }
+
+    public void createsDetectionRule() throws IOException, InterruptedException {
+        String creds = username + ":" + password;
+        String basicAuthPayload = "Basic " + Base64.getEncoder().encodeToString(creds.getBytes());
+
+        for (int retries = 0; ; retries++) {
+            String jsonstr = "{\"rule_id\": \"123\" , \"risk_score\": 7, \"description\": \"My description\", \"interval\": \"10s\", \"name\": \"Rule Name\", \"severity\": \"low\", \"type\": \"query\", \"from\": \"now-17520h\", \"index\": [\"auditbeat-upgrade-*\"], \"query\": \"*:*\", \"language\": \"kuery\", \"enabled\": true }";
```
Comment (MadameSheema) [inline ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java:109]: @liza-mae I'm not quite happy with this solution. Is there any easy way to have this json on a separate file? Thanks :)

Comment (liza-mae) [reply ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java]: @MadameSheema yes you can use a separate file.  You can place json files in ```ci/upgrade/buildSrc/src/main/resources```.  There is an example in:   
https://github.com/elastic/elastic-stack-testing/blob/7a11b7b943ae81d2c85b4a558d027070a1886d1f/ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UpgradeAssistantApi.java#L211

File: ci/upgrade/ess/build.gradle [hunk 2]
```diff
@@ -4,6 +4,7 @@ import org.estf.gradle.EditEssDeployment
 import org.estf.gradle.ShutdownEssDeployment
 import org.estf.gradle.UpgradeEssDeployment
 import org.estf.gradle.UploadData
+import org.estf.gradle.UploadDataSecuritySolution
```
Comment (liza-mae) [inline ci/upgrade/ess/build.gradle:7]: This should be: ```import org.estf.gradle.UploadSecuritySolutionData```

File: ci/upgrade/ess/build.gradle [hunk 3]
```diff
@@ -225,6 +226,16 @@ if (rootProject.createVersion && upgrade_versions ) {
             }
         }
 
+        tasks.create("upload_security_solution_data" + i, UploadDataSecuritySolution) {
```
Comment (liza-mae) [inline ci/upgrade/ess/build.gradle:229]: This should be: ```tasks.create("upload_security_solution_data" + i, UploadSecuritySolutionData) {```

File: ci/upgrade/ess/build.gradle [hunk 4]
```diff
@@ -263,7 +274,8 @@ if (rootProject.createVersion && upgrade_versions ) {
         }
 
         ess_upgrade.dependsOn "ess_upgrade" + i
-        tasks.findByName("upgrade_assistant" + i).mustRunAfter("upload_data" + i)
+        tasks.findByName("upload_security_solution_data" + i).mustRunFater("upload_data" + i)
```
Comment (liza-mae) [inline ci/upgrade/ess/build.gradle:277]: This should be: ```tasks.findByName("upload_security_solution_data" + i).mustRunAfter("upload_data" + i)```

File: ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+    @Input
+    public String kbnBaseUrl;
+
+    @Input
+    public String username;
+
+    @Input
+    public String password;
+
```
Comment (liza-mae) [inline ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java:52]: It appears that this data is not applicable to versions 5.x and 6.x, so let's only run them on 7.x or greater.

Please add the following input 
``` 
@Input 
public String version;

@Input
public String upgradeVersion;
```

File: ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    final private int MAX_RETRIES = 5;
+
+    @TaskAction
+    public void run() throws IOException, InterruptedException {
```
Comment (liza-mae) [inline ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java:56]: It appears that this data is not applicable to versions 5.x and 6.x, so let's only run them on 7.x or greater.

Please replace with the following version check: 
``` 
    RestApi api = new RestApi(username, password, version, upgradeVersion);
    int majorVersion = api.setMajorVersion();
    if (majorVersion > 6) {
        createsSiemSignalsIndex();
        createsDetectionRule();
        createsDocumentIndex();
        createsDocumentToGenerateAlert();
}

File: ci/upgrade/ess/build.gradle [hunk 7]
```diff
@@ -225,6 +226,16 @@ if (rootProject.createVersion && upgrade_versions ) {
             }
         }
 
+        tasks.create("upload_security_solution_data" + i, UploadDataSecuritySolution) {
+            doFirst {
+                println("******* SECTION: UPLOAD SECURITY SOLUTION DATA *******")
+                esBaseUrl = rootProject.props.elasticsearch_url
+                kbnBaseUrl = rootProject.props.kibana_url
+                username = rootProject.props.es_username
+                password = rootProject.props.es_password
```
Comment (liza-mae) [inline ci/upgrade/ess/build.gradle:235]: It appears that this data is not applicable to versions 5.x and 6.x, so let's only run them on 7.x or greater.
Please add the following options to your task: 
```
  version = originating_version
  upgradeVersion = upgrade_version
```

**Review by liza-mae — CHANGES_REQUESTED** [review]
Looks great, just a few changes requested.

File: ci/upgrade/ess/build.gradle (hunk 2 above)
Comment (MadameSheema) [reply ci/upgrade/ess/build.gradle:7]: True I did I typo here.

**Comment by MadameSheema** [comment]
@liza-mae I  have updated the PR with the required changes, can you please take a look at it again when you have the chance? Thanks :)

File: ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+            }
+        }
+    }
+
+    public void createsDetectionRule() throws IOException, InterruptedException {
+        String creds = username + ":" + password;
+        String basicAuthPayload = "Basic " + Base64.getEncoder().encodeToString(creds.getBytes());
+
+        for (int retries = 0; ; retries++) {
+            String file = "buildSrc/src/main/resources/detectionRule.json"
```
Comment (liza-mae) [inline ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java:120]: Add missing ```;``` at the end of line

File: ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+            String jsonStr = new String(Files.readAllBytes(Paths.get(file)))
```
Comment (liza-mae) [inline ci/upgrade/buildSrc/src/main/java/org/estf/gradle/UploadSecuritySolutionData.java:121]: Add missing ```;``` at the end of line

**Review by liza-mae — CHANGES_REQUESTED** [review]
Looks good, just a couple lines are missing an ending colon.

**Comment by MadameSheema** [comment]
Thanks for the review! Changes done @liza-mae

**Review by liza-mae — APPROVED** [review]
LGTM, thanks @MadameSheema

**Pull request merged**